tsdoc/src/parser/NodeParser.ts

```diff
@@ -1826,105 +1821,268 @@ export class NodeParser {
+  private _parseXmlElement(tokenReader: TokenReader): DocNode {
     tokenReader.assertAccumulatedSequenceIsEmpty();
-    const marker: number = tokenReader.createMarker();
+    const startMarker: number = tokenReader.createMarker();
 
     // Read the "<" delimiter
-    const lessThanToken: Token = tokenReader.readToken();
-    if (lessThanToken.kind !== TokenKind.LessThan) {
-      // This would be a parser bug -- the caller of _parseHtmlStartTag() should have verified this while
+    const startTagLessThanToken: Token = tokenReader.readToken();
+    if (startTagLessThanToken.kind !== TokenKind.LessThan) {
+      // This would be a parser bug -- the caller of _parseXmlElement() should have verified this while
       // looking ahead
-      throw new Error('Expecting an HTML tag starting with "<"');
+      throw new Error('Expecting an XML tag starting with "<"');
     }
 
-    // NOTE: CommonMark does not permit whitespace after the "<"
-
-    const openingDelimiterExcerpt: TokenSequence = tokenReader.extractAccumulatedSequence();
+    // Skip over opening "<"
+    const startTagOpeningDelimiterExcerpt: TokenSequence = tokenReader.extractAccumulatedSequence();
+    const startTagNameMarker: number = tokenReader.createMarker();
 
     // Read the element name
-    const nameExcerpt: ResultOrFailure<TokenSequence> = this._parseHtmlName(tokenReader);
-    if (isFailure(nameExcerpt)) {
+    const startTagNameExcerpt: ResultOrFailure<TokenSequence> = this._parseXmlName(tokenReader);
+    if (isFailure(startTagNameExcerpt)) {
       return this._backtrackAndCreateErrorForFailure(
         tokenReader,
-        marker,
-        'Invalid HTML element: ',
-        nameExcerpt
+        startTagNameMarker,
+        'Invalid XML tag name: ',
+        startTagNameExcerpt
       );
     }
 
-    const spacingAfterNameExcerpt: TokenSequence | undefined = this._tryReadSpacingAndNewlines(tokenReader);
-
-    const htmlAttributes: DocHtmlAttribute[] = [];
+    const spacingAfterStartTagNameExcerpt: TokenSequence | undefined = this._tryReadSpacingAndNewlines(
+      tokenReader
+    );
+    const xmlAttributes: DocXmlAttribute[] = [];
 
     // Read the attributes until we see a ">" or "/>"
     while (tokenReader.peekTokenKind() === TokenKind.AsciiWord) {
+      const attributeMarker: number = tokenReader.createMarker();
       // Read the attribute
-      const attributeNode: ResultOrFailure<DocHtmlAttribute> = this._parseHtmlAttribute(tokenReader);
+      const attributeNode: ResultOrFailure<DocXmlAttribute> = this._parseXmlAttribute(tokenReader);
       if (isFailure(attributeNode)) {
         return this._backtrackAndCreateErrorForFailure(
           tokenReader,
-          marker,
-          'The HTML element has an invalid attribute: ',
+          attributeMarker,
+          'Invalid XML attribute: ',
           attributeNode
         );
       }
 
-      htmlAttributes.push(attributeNode);
+      xmlAttributes.push(attributeNode);
```

Comment:
Not sure about HTML, but I'm pretty sure the XML spec forbids duplicated attributes (`<element a="1" a="2" />`). Should we validate that?